components/masking_functions/src/udf/udf_utils_string.cc

```diff
@@ -0,0 +1,207 @@
+/* Copyright (c) 2018, 2019 Francisco Miguel Biete Banon. All rights reserved.
+   Copyright (c) 2023 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   51 Franklin Street, Suite 500, Boston, MA 02110-1335 USA */
+
+#include "components/masking_functions/include/component.h"
+#include "components/masking_functions/include/udf/udf_utils.h"
+
+#include "components/masking_functions/include/udf/udf_utils_string.h"
+
+#include <algorithm>
+#include <cctype>
+#include <functional>
+#include <locale>
+
+#include "m_ctype.h"
+#include "my_sys.h"
+
+#include <mysql/components/services/mysql_string.h>
+extern REQUIRES_SERVICE_PLACEHOLDER(mysql_string_converter);
+extern REQUIRES_SERVICE_PLACEHOLDER(mysql_string_character_access);
+extern REQUIRES_SERVICE_PLACEHOLDER(mysql_string_factory);
+
+namespace mysql {
+namespace plugins {
+std::string &ltrim(std::string &s) {
+  s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](unsigned char c) {
```

Comment:
there is boost/algorithm/string/trim.hpp
https://www.boost.org/doc/libs/1_82_0/boost/algorithm/string/trim.hpp